wasmtime-api/src/types.rs

```diff
@@ -150,7 +150,13 @@ impl FuncType {
                 .iter()
                 .map(|p| AbiParam::new(p.get_cranelift_type()))
                 .collect::<Vec<_>>();
-            params.insert(0, AbiParam::special(types::I64, ArgumentPurpose::VMContext));
+            let ptr_ty = match HOST.pointer_width().unwrap() {
+                target_lexicon::PointerWidth::U16 => panic!("16-bit pointers not supported"),
+                target_lexicon::PointerWidth::U32 => types::I32,
```

Comment:
Just for my understanding: was the assumption so far that the callee is always 64-bit, but the caller could possibly be 32-bit? Or was the possibility of 32-bit simply neglected so far?

Reply:
I believe 32-bit was neglected so far. Cranelift doesn't support 32-bit yet, so it hasn't been a priority here, but it will in the future.